src/main/java/WayofTime/alchemicalWizardry/common/summoning/meteor/MeteorParadigm.java

```diff
@@ -143,43 +164,9 @@ public void createMeteorImpact(World world, int x, int y, int z, boolean[] flags
                         continue;
                     }
 
-                    int randNum = world.rand.nextInt(totalMeteorWeight);
-
-                    boolean hasPlacedBlock = false;
-
-                    for (MeteorParadigmComponent mpc : componentList) {
-                        randNum -= mpc.getChance();
-
-                        if (randNum < 0) {
-                            ItemStack blockStack = mpc.getValidBlockParadigm();
-                            if (blockStack != null && blockStack.getItem() instanceof ItemBlock) {
-                                ((ItemBlock) blockStack.getItem()).placeBlockAt(
-                                        blockStack,
-                                        null,
-                                        world,
-                                        x + i,
-                                        y + j,
-                                        z + k,
-                                        0,
-                                        0,
-                                        0,
-                                        0,
-                                        blockStack.getItemDamage());
-                                if (AlchemicalWizardry.isGregTechLoaded)
-                                    setGTOresNaturalIfNeeded(world, x + i, y + j, z + k);
-                                world.markBlockForUpdate(x + i, y + j, z + k);
-                                hasPlacedBlock = true;
-                                break;
-                            }
-                            // world.setBlock(x + i, y + j, z + k,
-                            // Block.getBlockById(Item.getIdFromItem(blockStack.getItem())), blockStack.getItemDamage(),
-                            // 3);
-                            // hasPlacedBlock = true;
-                            // break;
-                        }
-                    }
-
-                    if (!hasPlacedBlock) {
+                    if (world.rand.nextInt(100) >= fillerChance) {
```

Comment:
`if (fillerChance <= 0 || world.rand.nextInt(100) >= fillerChance)` for short-circuiting